### PR TITLE
EIP1-1878 Write print requests to output stream in agreed PSP format

### PR DESCRIPTION
**Pull request by Neil-Massey**
Writes the print requests in a format agreed with the printer to an output stream that will be included in a zip file sent to the printer's SFTP server

### CODE REVIEW

File: src/main/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsFileProducer.kt [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                deliveryOption.toString(),
+                photo,
+                deliveryName,
+                deliveryStreet,
+                Optional.ofNullable(deliverypProperty).orElse(""),
+                Optional.ofNullable(deliveryLocality).orElse(""),
+                Optional.ofNullable(deliveryTown).orElse(""),
+                Optional.ofNullable(deliveryArea).orElse(""),
+                Optional.ofNullable(deliveryPostcode).orElse(""),
+                eroNameEn,
```
Comment (adacayi) [inline src/main/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsFileProducer.kt:106]: How about this approach, @Neil-Massey ?

                deliverypProperty ?: "",
                deliveryLocality ?: "",
                deliveryTown ?: "",
                deliveryArea ?: "",
                deliveryPostcode ?: "",

File: src/main/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsFileProducer.kt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                cardSurname,
+                cardVersion,
+                cardNumber,
+                certificateLanguage.toString(),
+                certificateFormat.toString(),
+                deliveryOption.toString(),
+                photo,
+                deliveryName,
+                deliveryStreet,
+                Optional.ofNullable(deliverypProperty).orElse(""),
```
Comment (Neil-Massey) [inline src/main/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsFileProducer.kt:101]: Need to verify whether these can be null - as the PrintRequest is generated it's not helping with nulls... and test data may not be fully representative regarding what is expected in full flow

File: src/test/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsFileProducerTest.kt [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            """.trimIndent()
+        )
+    }
+
+    @Test
+    fun `should write file to stream given print request with optional properties missing`() {
+        // Given
+        val outputStream = ByteArrayOutputStream()
+        val printRequests = listOf(
+            buildPrintRequest(
```
Comment (Neil-Massey) [inline src/test/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsFileProducerTest.kt:85]: PrintRequest.yaml defines following:
required:
  - requestId
  - issuingAuthorityEn
  - issuingAuthorityCy
  - issueDate
  - suggestedExpiryDate
  - requestDateTime
  - cardFirstname
  - cardSurname
  - cardVersion
  - cardNumber
  - certificateLanguage
  - certificateFormat
  - deliveryOption
  - photo
  - deliveryName
  - deliveryStreet
  - deliveryPostcode
  - eroNameEn
  - eroPhoneNumberEn
  - eroEmailAddressEn
  - eroWebsiteEn
  - eroDeliveryStreetEn
  - eroDeliveryPostcodeEn

File: src/test/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsFileProducerTest.kt [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                requestId = "627ab400-5ae2-4cc5-9e91-5c050e43e4c1",
+                issuingAuthorityEn = "Lake Deedra",
+                issuingAuthorityCy = "Bradtketon",
```
Comment (Neil-Massey) [inline src/test/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsFileProducerTest.kt:88]: I'm not sure how `issuingAuthorityCy` can be required when there may not be a Welsh ERO - @NathanRussellValtech

File: src/test/kotlin/uk/gov/dluhc/printapi/testsupport/testdata/model/PrintRequestBuilder.kt [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+import uk.gov.dluhc.printapi.testsupport.testdata.DataFaker.Companion.faker
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildElectoralRegistrationOffice
+import uk.gov.dluhc.printapi.testsupport.testdata.getAMongoDbId
+import java.time.Instant
+import java.util.Date
+
+fun buildPrintRequest(
+    eroEnglish: ElectoralRegistrationOffice = buildElectoralRegistrationOffice(),
+    eroWelsh: ElectoralRegistrationOffice? = buildElectoralRegistrationOffice(),
+    requestId: String = getAMongoDbId(),
```
Comment (adacayi) [inline src/test/kotlin/uk/gov/dluhc/printapi/testsupport/testdata/model/PrintRequestBuilder.kt:14]: Minor, @Neil-Massey.
We can use `aValidRequestId()` here.

**Review by adacayi — APPROVED** [review]
LGTM, @Neil-Massey , thanks.

**Pull request merged**